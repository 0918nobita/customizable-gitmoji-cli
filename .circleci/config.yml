version: 2.1

executors:
  node-lts:
    docker:
      - image: circleci/node:lts

jobs:
  install:
    executor: node-lts
    steps:
      - checkout
      - restore_cache:
          keys:
            - dependency-cache-{{ checksum "package-lock.json" }}
            - dependency-cache-
      - run: npm ci
      - save_cache:
          key: dependency-cache-{{ checksum "package-lock.json" }}
          paths:
            - node_modules
  test:
    executor: node-lts
    steps:
      - checkout
      - run: npm test

workflows:
  version: 2
  test:
    jobs:
      - install
      - test
