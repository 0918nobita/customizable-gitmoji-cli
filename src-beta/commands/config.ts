import Conf from 'conf'
import inquirer from 'inquirer'

// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'))

const config = new Conf()

export enum ConfigKeys {
  AUTO_ADD = 'auto_add',
  EMOJI_FORMAT = 'emoji_format',
  SIGNED_COMMIT = 'signed_commit',
  TITLE_MAX_LENGTH = 'title_max_length'
}

type Config = {
  [ConfigKeys.AUTO_ADD]: boolean
  [ConfigKeys.EMOJI_FORMAT]: 'emoji' | 'code'
  [ConfigKeys.SIGNED_COMMIT]: boolean
  [ConfigKeys.TITLE_MAX_LENGTH]: number
}

export function getConfig(): Config {
  return {
    [ConfigKeys.AUTO_ADD]: config.get(ConfigKeys.AUTO_ADD, false),
    [ConfigKeys.EMOJI_FORMAT]: config.get(ConfigKeys.EMOJI_FORMAT, 'code'),
    [ConfigKeys.SIGNED_COMMIT]: config.get(ConfigKeys.SIGNED_COMMIT, false),
    [ConfigKeys.TITLE_MAX_LENGTH]: config.get(ConfigKeys.TITLE_MAX_LENGTH, 48)
  }
}

export default async function() {
  const answer = await inquirer.prompt([
    {
      name: ConfigKeys.AUTO_ADD,
      message: 'Enable automatic "git add ."',
      type: 'confirm'
    },
    {
      name: ConfigKeys.EMOJI_FORMAT,
      message: 'Select how emojis should be used in commits',
      type: 'list',
      choices: [
        { name: ':smile:', value: 'code' },
        { name: 'ğŸ˜„', value: 'emoji' }
      ]
    },
    {
      name: ConfigKeys.SIGNED_COMMIT,
      message: 'Enable signed commits',
      type: 'confirm'
    },
    {
      name: ConfigKeys.TITLE_MAX_LENGTH,
      message: 'Max length of commit title',
      type: 'number',
      validate: number => number > 0 && number <= 72,
      default: 48
    }
  ])
  config.set(ConfigKeys.AUTO_ADD, answer[ConfigKeys.AUTO_ADD])
  config.set(ConfigKeys.EMOJI_FORMAT, answer[ConfigKeys.EMOJI_FORMAT])
  config.set(ConfigKeys.SIGNED_COMMIT, answer[ConfigKeys.SIGNED_COMMIT])
  config.set(ConfigKeys.TITLE_MAX_LENGTH, answer[ConfigKeys.TITLE_MAX_LENGTH])
}
