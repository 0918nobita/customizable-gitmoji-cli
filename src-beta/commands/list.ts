import gitmojis from '../gitmojis'

import parseLine from '../parseLine'

export default function() {
  parseLine(gitmojis)
}
