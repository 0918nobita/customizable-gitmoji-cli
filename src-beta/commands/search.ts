import chalk from 'chalk'
import consola from 'consola'

import gitmojis from '../gitmojis'
import parseLine from '../parseLine'

export default function(query?: string) {
  if (!query) return
  if (
    !parseLine(
      gitmojis.filter(({ name, description }) =>
        `${name}${description}`.includes(query)
      ),
      line => {
        const indexOf = line.indexOf(query)
        const { length } = query
        return [
          line.substring(0, indexOf),
          chalk.black.bgGreen.bold(line.substr(indexOf, length)),
          line.substring(indexOf + length)
        ].join('')
      }
    )
  )
    consola.error(`No gitmoji is found query "${query}"`)
}
