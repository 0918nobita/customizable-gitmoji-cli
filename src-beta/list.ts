import chalk from 'chalk'

import gitmojis from './gitmojis'

export default function() {
  const longest = gitmojis
    .map(({ code }) => code.length)
    .reduce((acc, cur) => Math.max(acc, cur))
  gitmojis.forEach(({ emoji, code, description }) => {
    console.log(
      [
        emoji,
        ' - ',
        chalk.blue(code),
        ''.padStart(longest - code.length + 1, ' '),
        'â”‚ ',
        description
      ].join('')
    )
  })
}
