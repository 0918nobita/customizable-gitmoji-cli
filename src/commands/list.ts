import gitmojis from '../gitmojis'

import parseLines from '../parseLines'

export function list(array: Gitmoji[]) {
  const lines = parseLines(array)
  // eslint-disable-next-line no-console
  if (lines) lines.forEach(line => console.log(line))
}

export default function() {
  list(gitmojis)
}
