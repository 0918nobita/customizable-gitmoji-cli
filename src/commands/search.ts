import chalk from 'chalk'
import consola from 'consola'

import gitmojis from '../gitmojis'
import parseLines from '../parseLines'

export function search(array: Gitmoji[], query?: string) {
  if (!query) {
    consola.error(`No query for search`)
    return false
  }
  const lines = parseLines(
    array.filter(({ name, description }) =>
      `${name}${description}`.includes(query)
    )
  )
  if (lines) {
    // eslint-disable-next-line no-console
    lines.forEach(line => console.log(line))
    return true
  }
  consola.error(`No gitmoji is found query "${query}"`)
  return false
}

export default function(query?: string) {
  search(gitmojis, query)
}
