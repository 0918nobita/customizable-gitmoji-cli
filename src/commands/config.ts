import consola from 'consola'
import chalk from 'chalk'
import inquirer from 'inquirer'
import { ConfigKeys, setConfig, defaults, getConfigs } from '../config'

// eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'))

export async function list() {
  const configs = await getConfigs()
  const longest = Object.keys(configs)
    .map(c => c.length)
    .reduce((accumulator, current) => Math.max(accumulator, current))
  Object.entries(configs).forEach(({ 0: key, 1: value }) => {
    consola.info(
      [
        key,
        ''.padStart(longest - key.length + 1, ' '),
        ': ',
        chalk.yellow(String(value))
      ].join('')
    )
  })
}

export default async function() {
  const answer = await inquirer.prompt([
    {
      name: ConfigKeys.AUTO_ADD,
      message: 'Enable automatic "git add ."',
      type: 'confirm',
      default: defaults[ConfigKeys.AUTO_ADD]
    },
    {
      name: ConfigKeys.EMOJI_FORMAT,
      message: 'Select how emojis should be used in commits',
      type: 'list',
      choices: [
        { name: ':smile:', value: 'code' },
        { name: 'ğŸ˜„', value: 'emoji' }
      ],
      default: defaults[ConfigKeys.EMOJI_FORMAT]
    },
    {
      name: ConfigKeys.SIGNED_COMMIT,
      message: 'Enable signed commits',
      type: 'confirm',
      default: defaults[ConfigKeys.SIGNED_COMMIT]
    },
    {
      name: ConfigKeys.TITLE_MAX_LENGTH,
      message: 'Max length of commit title',
      type: 'number',
      validate: number => number > 0 && number <= 72,
      default: defaults[ConfigKeys.TITLE_MAX_LENGTH]
    }
  ])
  setConfig(ConfigKeys.AUTO_ADD, answer[ConfigKeys.AUTO_ADD])
  setConfig(ConfigKeys.EMOJI_FORMAT, answer[ConfigKeys.EMOJI_FORMAT])
  setConfig(ConfigKeys.SIGNED_COMMIT, answer[ConfigKeys.SIGNED_COMMIT])
  setConfig(ConfigKeys.TITLE_MAX_LENGTH, answer[ConfigKeys.TITLE_MAX_LENGTH])
}
