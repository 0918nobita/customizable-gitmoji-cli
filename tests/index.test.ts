import consola from 'consola'

import { getConfig, setConfig, ConfigKeys } from '~/config'
import { parseLine } from '~/parseLines'

import { list } from '~/commands/list'
import { search } from '~/commands/search'

consola.mockTypes(() => jest.fn)

describe('config module', () => {
  it('should match for AUTO ADD', () => {
    setConfig(ConfigKeys.AUTO_ADD, true)
    expect(getConfig(ConfigKeys.AUTO_ADD)).toMatchSnapshot()
  })

  it('should match for EMOJI FORMAT', () => {
    setConfig(ConfigKeys.EMOJI_FORMAT, 'emoji')
    expect(getConfig(ConfigKeys.EMOJI_FORMAT)).toMatchSnapshot()
  })

  it('should match for SIGNED COMMIT', () => {
    setConfig(ConfigKeys.SIGNED_COMMIT, false)
    expect(getConfig(ConfigKeys.SIGNED_COMMIT)).toMatchSnapshot()
  })

  it('should match for TITLE MAX LENGTH', () => {
    setConfig(ConfigKeys.TITLE_MAX_LENGTH, 48)
    expect(getConfig(ConfigKeys.TITLE_MAX_LENGTH)).toMatchSnapshot()
  })
})

const gitmojis = [
  {
    emoji: 'ðŸŽ¨',
    entity: '&#x1f3a8;',
    code: ':art:',
    description: 'Improving structure / format of the code.',
    name: 'art'
  },
  {
    emoji: 'ðŸ”¥',
    entity: '&#x1f525;',
    code: ':fire:',
    description: 'Removing code or files.',
    name: 'fire'
  }
]

describe('commands', () => {
  const log = jest.spyOn(console, 'log')
  log.mockImplementation(x => x)

  describe('list', () => {
    it('should match for list', () => {
      list(gitmojis)
      expect(console.log).toHaveBeenCalledTimes(gitmojis.length)
      for (let i = 0; i < gitmojis.length; i += 1)
        expect(log.mock.calls[i][0]).toMatchSnapshot()
      log.mockReset()
      log.mockRestore()
    })
  })

  describe('search', () => {
    it('should false for no query', () => {
      expect(search(gitmojis)).toEqual(false)
    })
    it('should match for correct query', () => {
      search(gitmojis, 'Removing')
      expect(log.mock.calls[0]).toMatchSnapshot()
      log.mockReset()
      log.mockRestore()
    })
  })
})

describe('parse line', () => {
  it('should match for parseLine', () => {
    expect(parseLine(gitmojis[0], 3)).toMatchSnapshot()
  })
})
