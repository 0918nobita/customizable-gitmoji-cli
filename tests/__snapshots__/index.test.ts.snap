// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands list should match for list 1`] = `"ðŸŽ¨ - [34m:art:[39m  â”‚ Improving structure / format of the code."`;

exports[`commands list should match for list 2`] = `"ðŸ”¥ - [34m:fire:[39m â”‚ Removing code or files."`;

exports[`commands search should match for correct query 1`] = `undefined`;

exports[`config module should match for AUTO ADD 1`] = `true`;

exports[`config module should match for EMOJI FORMAT 1`] = `"emoji"`;

exports[`config module should match for SIGNED COMMIT 1`] = `false`;

exports[`config module should match for TITLE MAX LENGTH 1`] = `48`;

exports[`parse line should match for parseLine 1`] = `"ðŸŽ¨ - [34m:art:[39mâ”‚ Improving structure / format of the code."`;
